core/lib/l1_contract_interface/src/i_executor/structures/commit_batch_info.rs

```diff
@@ -101,6 +109,72 @@ impl<'a> Tokenizable for CommitBatchInfo<'a> {
                 ),
                 // `systemLogs`
                 Token::Bytes(serialize_commitments(&self.0.header.system_logs)),
+            ]
+        }
+    }
+
+    pub fn into_tokens_calldata(&self) -> Token {
+        let mut tokens = self.as_tokens_base();
+        let mut pubdata = self
+            .0
+            .header
+            .pubdata_input
+            .clone()
+            .unwrap_or(self.0.construct_pubdata());
+
+        let kzg_settings = load_kzg_settings();
+        let mut blob = pubdata.clone();
+        blob.resize(ZK_SYNC_BYTES_PER_BLOB, 0u8);
+        let blob_commitment = KzgInfo::new(&kzg_settings, &blob).to_blob_commitment();
+
+        pubdata.extend(blob_commitment);
+
```

Comment:
```suggestion
        let mut pubdata = vec![PUBDATA_SOURCE_CALLDATA];
        pubdata.extend_from_slice(
            self.0
                .header
                .pubdata_input
                .as_ref()
                .unwrap_or(&self.0.construct_pubdata()),
        );

        let kzg_settings = load_kzg_settings();
        let mut blob = pubdata.clone();
        blob.resize(ZK_SYNC_BYTES_PER_BLOB, 0u8);
        let blob_commitment = KzgInfo::new(&kzg_settings, &blob).to_blob_commitment();

        pubdata.extend(blob_commitment);
```

Still, i think that this can avoid both the `insert` and another `clone`.